src/components/business/AddIdentity.tsx

```diff
@@ -0,0 +1,37 @@
+import React from "react";
+
+import { HttpVerbs } from "../../@types/HttpVerbs";
+import { IdentityTypes } from "../../@types/Identity";
+import { useCookies } from "../../hooks/useCookies";
+import { fetchJson } from "../../utils/fetchJson";
+
+interface AddIdentityProps {
+  type: IdentityTypes;
+  children: (props: {
+    addIdentity: (value: string) => Promise<Response>;
+  }) => JSX.Element;
+}
+
+export const AddIdentity: React.FC<AddIdentityProps> = ({ type, children }) => {
+  const { data, error } = useCookies();
+
+  if (error) {
+    return <div>Failed to load</div>;
+  }
+
+  if (!data) {
+    return null;
+  }
+
+  const addIdentity = (value: string): Promise<Response> => {
+    const body = {
+      userId: data.userId,
+      type,
+      value,
+    };
+
+    return fetchJson("/api/account", HttpVerbs.POST, body);
```

Comment:
Should we assume that it will always return json and already do `response.json()` there?
It's really a question, I have no prior opinion on this

Reply:
I guess it would be more flexible to not return a JSON, as we would be able to throw specific errors based on what kind of fetch we want to achieve. Kind of thinking out loud, is that the kind of interrogation you had in mind?

Reply:
Yes, that and, whose job is it to deal with errors?

Reply:
Next internals I guess? Or do you want us to handle errors and fallback components ourselves?

Reply:
Imagine if we want to have flash messages that pop up when an error occurs, we would need to catch them (for small errors in my mind, if all goes to hell, then next to the rescue)

Reply:
So we keep this as it is, right?

Reply:
yup, for now